Assets/Scripts/Models/Character.cs

```diff
@@ -1017,9 +999,9 @@ private bool CheckForJobMaterials()
 
     private bool WalkingToUsableInventory()
     {
-        bool destHasInventory = pathAStar != null && pathAStar.EndTile() != null && pathAStar.EndTile().Inventory != null;
+        bool destHasInventory = movementPath != null && movementPath.Last() != null && movementPath.Last().Inventory != null;
```

Comment:
@sboigelot That would probably be a good change and would probably stop the log spam, but I think the deeper issue is that somehow two characters are getting the same job in odd circumstances.